rust/chains/hyperlane-ethereum/src/mailbox.rs

```diff
@@ -225,10 +225,23 @@ where
                 .await?
                 .saturating_add(U256::from(GAS_ESTIMATE_BUFFER))
         };
+        let chain_id = match self.provider.get_chainid().await {
+            Ok(chainId) => chainId.as_u32(),
+            // Couldn't get chainId, assume not 1559
+            Err(_) => return Ok(tx.gas(gas_limit)),
+        };
         let Ok((max_fee, max_priority_fee)) = self.provider.estimate_eip1559_fees(None).await else {
             // Is not EIP 1559 chain
             return Ok(tx.gas(gas_limit))
         };
+        let max_priority_fee =
+            if KnownHyperlaneDomain::try_from(chain_id)? == KnownHyperlaneDomain::Polygon {
+                let min_polygon_fee = ethers::utils::parse_units("31", "gwei").unwrap().into();
```

Comment:
fwiw the condition is >= 30 gwei and not > 30 gwei

Also do we need to unwrap here?

This would probably be most preferable imo:
```
// 30 gwei
let min_polygon_fee = U256::from(30000000000);
```

or if you don't like that maybe
```
let min_polygon_fee = ethers::utils::parse_units("30", "gwei")?.into();
```

Reply:
I actually had your suggestion first (from codepilot!) and thought people liked the use of the utility.